### PR TITLE
check with cargo-audit in CI

### PR DESCRIPTION
**Pull request by arniu**
## PR Type

Other

## PR Checklist
Check your PR fulfills the following:

- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] A changelog entry has been made for the appropriate packages.
- [ ] Format code with the latest stable rustfmt

### CODE REVIEW

File: .github/workflows/audit.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Security audit
+
+on:
+  push:
+    paths: 
+      - '**/Cargo.toml'
+      - '**/Cargo.lock'
```
Comment (JohnTitor) [inline .github/workflows/audit.yml:7]: This doesn't make sense as we don't add Cargo.lock in this repo, right?

File: .github/workflows/audit.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  security_audit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
```
Comment (JohnTitor) [inline .github/workflows/audit.yml]:
```suggestion
      - uses: actions/checkout@v2
```

File: .github/workflows/audit.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      - uses: actions-rs/audit-check@v1
```
Comment (JohnTitor) [inline .github/workflows/audit.yml:16]: Shouldn't we generate lockfile before running check?

**Review by JohnTitor — COMMENTED** [review]
It seems this workflow is just copied/pasted from actions-rs/audit-check's README, do you have any context?
I know, cargo-audit is a great tool and it's _ideal_ to use here but I'm not sure if we have enough bandwidth to deal with issues from cargo-audit.

**Comment by robjtede** [comment]
It's worth noting that a deps.rs badge on the readme (+ dependabot PRs maybe) would be plenty enough to catch insecure deps. I'm not convinced about adding this into the CI.

File: .github/workflows/audit.yml (hunk 1 above)
Comment (arniu) [reply .github/workflows/audit.yml:7]: Yes, we didn't. But just in case.

File: .github/workflows/audit.yml (hunk 2 above)
Comment (arniu) [reply .github/workflows/audit.yml]: My bad! I'll fix it.

File: .github/workflows/audit.yml (hunk 3 above)
Comment (arniu) [reply .github/workflows/audit.yml:16]: `actions-rs/audit-check` will generate lockfile before checking.

**Comment by arniu** [comment]
> It's worth noting that a deps.rs badge on the readme (+ dependabot PRs maybe) would be plenty enough to catch insecure deps. I'm not convinced about adding this into the CI.

Dependabot is fine, except too annoying.

**Comment by JohnTitor** [comment]
So, no one from the team acks this change, I'm in favor of just closing this.
@actix/contributors If anyone is interested in introducing `cargo-audit`, feel free to leave a comment :) I'll keep this open in a few days then close it.

**Comment by robjtede** [comment]
No joy. FWIW, I'm intending to add a deps.rs badge to the README soon when we've implemented the "latest published crate status" feature.

**Pull request closed without merging**